### PR TITLE
[Networking] Basic message server and driver install script

### PR DESCRIPTION
**Pull request by Alramech**
- currently requires manual ip input, defualt localhost

- client closes after sending message, to be changed to persistent

### CODE REVIEW

File: networking/client.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import socket
+
+s = socket.socket(
```
Comment (yizow) [inline networking/client.py]: why split this into multiple lines?

File: networking/client.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    socket.AF_INET, socket.SOCK_STREAM)
+
+s.connect(('localhost', 3141))
```
Comment (yizow) [inline networking/client.py]: if 3141 is going to be a constant, bind it to a global variable after your import statements.

File: networking/client.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+msg = raw_input('message: ')
+while msg != '':
+    sent = s.send(msg)
+    msg = msg[sent:]
+s.shutdown(socket.SHUT_RDWR)
+s.close()
```
Comment (yizow) [inline networking/client.py]: add newline at end of file

File: networking/server.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import socket
+
+s = socket.socket(
```
Comment (yizow) [inline networking/server.py]: Same comment as above

File: networking/server.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    socket.AF_INET, socket.SOCK_STREAM)
+
+s.bind(('localhost', 3141))
```
Comment (yizow) [inline networking/server.py]: Same comment as above
Use global costants.

File: networking/server.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+s.listen(5)
```
Comment (yizow) [inline networking/server.py]: Why choose 5? Leaving it blank lets the system choose a "reasonable value"

File: networking/server.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+while 1:
+    (cs, addr) = s.accept()
+    msg = ''
+    rec = cs.recv(1024)
+    while rec:
+        msg += rec
+        rec = cs.recv(1024)
+    print(msg)
```
Comment (yizow) [inline networking/server.py]: New line at end of file

File: networking/client.py (hunk 3 above)
Comment (yizow) [inline networking/client.py]: Using `with s as socket.socket(...):` autoamtically calls close at end of block

File: networking/server.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import socket
+
+s = socket.socket(
+    socket.AF_INET, socket.SOCK_STREAM)
+
+s.bind(('localhost', 3141))
+s.listen(5)
+while 1:
+    (cs, addr) = s.accept()
```
Comment (yizow) [inline networking/server.py]: I recommend renaming `cs` to `conn`. More readable.

File: networking/client.py [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import socket
+
+s = socket.socket(
+    socket.AF_INET, socket.SOCK_STREAM)
+
+s.connect(('localhost', 3141))
+
+msg = raw_input('message: ')
```
Comment (yizow) [inline networking/client.py]: we're using python3.4 currently. `raw_input` is deprecated and has been renamed to `input`

File: networking/client.py [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+while msg != '':
+    sent = s.send(msg)
```
Comment (yizow) [inline networking/client.py]: This will no longer work in python3. msg must support buffer interface, so use `bytes(msg)`

**Review by yizow — CHANGES_REQUESTED** [review]
Main issue: Update to python 3.4

This is a lot more barebones that I was hoping for :-/

Have you tested this running on different machines? 
Or can you spin up more than just two processes: Simulate a (Driver station, field control, robot) network?
After making such a simulation, how would you verify it works? Then, write a test script that does that verfication.

File: networking/server.py (hunk 6 above)
Comment (Alramech) [reply networking/server.py]: That's a 3.5 thing, not a 3.4. 5 is usually the system maximum value.

File: networking/server.py [hunk 11]
```diff
@@ -1,15 +1,16 @@
 import socket
 
-s = socket.socket(
-    socket.AF_INET, socket.SOCK_STREAM)
+HOST = 'localhost'
```
Comment (yizow) [inline networking/server.py]: why are port and host in the opposite ordering here ._.

File: networking/server.py [hunk 12]
```diff
@@ -1,15 +1,16 @@
...
-    print(msg)
\ No newline at end of file
+with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+    s.bind((HOST, PORT))
+    s.listen(5)
+    while 1:
+        conn, addr = s.accept()
+        msg = bytes()
+        rec = conn.recv(1024)
+        while rec != bytes():
```
Comment (yizow) [inline networking/server.py]: i think `while rec` works, and is cleaner. You should check. I'm going off the python doc example which is `if not data: break`

**Review by yizow — CHANGES_REQUESTED** [review]

File: networking/server.py [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+            msg += rec
+            rec = conn.recv(1024)
+        print(msg.decode('UTF8'))
```
Comment (yizow) [inline networking/server.py:25]: Do we have a newline at end of file?

**Review by yizow — CHANGES_REQUESTED** [review]

**Review by yizow — APPROVED** [review]
Good work. In future patches, i'd like to see: 
1. Some sort of test. Send things, print things to a file, examine file. AUTOMATED.
2. More than 2 clients.
3. An interactive mode where we can send user-input messages. This will make debugging and testing the performance of runtime/ dawn super super useful. 

Merge when you're ready.

**Pull request merged**